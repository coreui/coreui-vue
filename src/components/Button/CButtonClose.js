import { mergeData } from 'vue-functional-data-merge'
export default {
  name: 'CButtonClose',
  functional: true,
  render (h, { data, props, slots }) {
    const componentData = {
      staticClass: 'close',
      attrs: {
        type: props.type || 'button',
        'aria-label': props.ariaLabel || 'Close'
      }
    }
    if (!slots().default)
      componentData.domProps = { innerHTML: '&times;' }
    return h('button', mergeData(data, componentData), slots().default)
  }
}
