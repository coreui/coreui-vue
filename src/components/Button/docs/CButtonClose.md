### Button close (CButtonClose) [Functional]

This component behaviour is the same as html button element so you can add standard html attributes freely. By default component will output:

```html
<button aria-label="Close"
        class="close"
        type="button"
>
  &times;
</button>
```

By passing class or style attribute you are adding styles to 'close' class. By adding html attributes you overwrite them.

Default slot content will overwrite "&times;" icon.
