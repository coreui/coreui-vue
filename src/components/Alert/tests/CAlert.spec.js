import { mount } from '@vue/test-utils'
import Component from "../CAlert";
import CButtonClose from "../../Button/CButtonClose";

const ComponentName = 'CAlert'
const defaultWrapper = mount(Component, {
  propsData: {
    show: true
  },
  slots: {
    default: 'Dismissible Alert!'
  }
})

const customWrapper = mount(Component, {
  propsData: {
    show: true,
    fade: true,
    dismissible: true,
    variant: 'success'
  },
  slots: {
    default: 'Dismissible Alert!'
  },
  scopedSlots: {
    default:`   <CButtonClose slot-scope="{ dismiss }"
                    @click="dismiss"
                    aria-label="close it"
                    style="color:red">
                    ok
                </CButtonClose>`
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
