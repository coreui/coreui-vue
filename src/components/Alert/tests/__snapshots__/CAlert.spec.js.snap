// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAlert renders correctly 1`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="alert alert-info"
  role="alert"
>
  <!---->
  Dismissible Alert!
</div>
`;

exports[`CAlert renders correctly 2`] = `
<div
  aria-atomic="true"
  aria-live="polite"
  class="alert alert-success alert-dismissible"
  role="alert"
>
  <cbuttonclose
    aria-label="close it"
    style="color: red;"
  >
    
                    ok
                
  </cbuttonclose>
  Dismissible Alert!
</div>
`;
